src/components/Form/formFunctions.ts

```diff
@@ -120,6 +120,59 @@ export const onChangeRadioGroup = (
     };
 };
 
+export const valueOnChangeRadioGroup = (
+    fieldSetter: FieldSetter,
+    errorSetter: ErrorSetter,
+    key: string
+): selectChangeEventHandler => {
+    return (event) => {
+        const input = event.target.value;
+        fieldSetter(key, input);
+        errorSetter(key, Errors.none);
+    };
+};
+
+export const onChangeDate = (
+    fieldSetter: FieldSetter,
+    errorSetter: ErrorSetter,
+    key: string,
+    value: Date | null
+): void => {
+    if (value === null) {
+        return;
+    }
+    const input = new Date(value);
+    const date = {
+        year: input?.getFullYear(),
+        month: input?.getMonth(),
+        day: input?.getDate(),
+    };
+    fieldSetter(key, value);
+    !(date.year && date.month && date.day)
+        ? errorSetter(key, Errors.invalid)
+        : errorSetter(key, Errors.none);
+};
+
+export const onChangeTime = (
+    fieldSetter: FieldSetter,
+    errorSetter: ErrorSetter,
+    key: string,
+    value: Date | null
+): void => {
+    if (value === null) {
+        fieldSetter(key, null);
+        return;
+    }
+    const input = new Date(value);
+    const time = {
+        hours: input?.getHours(),
+        minutes: input?.getMinutes(),
+    };
+    fieldSetter(key, value);
+    !(time.hours && time.minutes) && time.hours !== 0 && time.minutes !== 0
+        ? errorSetter(key, Errors.invalid)
+        : errorSetter(key, Errors.none);
```

Comment:
I would change this to an `if` `else` block instead of a ternary operator. You can also simplify the logic in this function in general after addressing the comment above on validating the date first: you no longer need to construct the `time` variable and can instead just call the corresponding `.getHours()` and `.getMinutes()` on the date object itself